connectors/fs2-connector/src/main/scala/zio/connect/fs2/FS2Connector.scala

```diff
@@ -0,0 +1,21 @@
+package zio.connect.fs2
+
+import cats.effect.kernel.Async
+import cats.effect.std.Dispatcher
+import zio._
+import zio.stream.ZStream
+
+trait FS2Connector {
+
+  def fromStream[F[_]: Dispatcher, R, A](
```

Comment:
Is there a way to get rid of that F[_] there?

Reply:
Let me clarify this. Do you want to have a fixed effect, e.g. RIO[R, Throwable] instead of `F[_]`, or to make it a parameter of `FS2Connector` trait?

Reply:
In the trait the R should be any. The implementations deal with dependencies.
I think we should not be providing support for F[_] in the api. We have to see how the custom FS2Stream will look like.

Maybe smth like this.
```
final case class FS2Stream(stream: fs2.Stream[zio.Task, A])

def fromStream[A](original: FS2Stream[A],
     queueSize: Int = 16
   )(implicit
     trace: Trace
   ): ZStream[Any, FS2Exception, A]
```

Reply:
But I think I should ask why don't we want to have an abstract `F`?
As I understand, your line of thought is that the user should take care of translating stream to `Task` before using `fromStream`, so the connector library wouldn't depend on an additional `cats-interop` library.

Reply:
I misunderstood how it would look like. We should not have a TF style api.
Regarding the user having to translate the stream to Task I think it's acceptable.

Reply:
I think it's fine either way. Types will infer a bit better without the type classes.

Reply:
Okay, in this case, I'll stick to `Task` instead of `F` and will remove `toStream` as Adrian suggested in the comment above.